api/python/provisioner/commands/__init__.py

```diff
@@ -751,9 +755,14 @@ def _update_component(component, targets=ALL_MINIONS):
 
 
 def _apply_provisioner_config(targets=ALL_MINIONS):
-    logger.info(f"Applying Provisioner config logic on {targets}")
-    StatesApplier.apply(["components.provisioner.config"], targets)
-
+    try:
+        logger.info(f"Applying Provisioner config logic on {targets}")
+        StatesApplier.apply(["components.provisioner.config"], targets)
+    except Exception:
+        logger.exception(
+            "Failed to appy Provisioner config logic on on {}".format(targets)
```

Comment:
it is an unnecessary redundancy / duplication : the very similar thing will be done in `try-except` block where that function is called. In current code we don't deal with exceptions inside functions but in outer logic instead.